deep_qa/training/trainer.py

```diff
@@ -319,7 +319,7 @@ def train(self):
         # so you may have left it above zero on accident.
         if self.validation_arrays is not None:
             kwargs['validation_data'] = self.validation_arrays
-        elif self.validation_split > 0.0:
+        elif self.validation_split > 0.0 and isinstance(self.training_arrays, tuple):
```

Comment:
Hmm actually i've re-thought about this and it seems to break quite heavily with the parameter popping idea and i'm not so sure it's a good idea - having parameters interact across classes is quickly going to get complicated.